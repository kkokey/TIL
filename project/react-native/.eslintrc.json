{
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "node": true,
    "jest": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ],
  "parser": "babel-eslint",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": 2018,
    "sourceType": "module"
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "no-console": "warn",
    "indent": [
      "error",
      2
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "semi": [
      "error",
      "always"
    ],
    "react/jsx-uses-react": "error",
    "react/jsx-uses-vars": "error",
    "react/prop-types": [
      0
    ]
  }
}
