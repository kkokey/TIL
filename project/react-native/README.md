## React-native Ex Project

#### react-native + mobx + jest

#### install
```bash
$yarn or yarn install
```

#### test
```bash
yarn test
```
