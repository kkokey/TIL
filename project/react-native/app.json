{
  "expo": {
    "name": "react-native",
    "slug": "react-native",
    "privacy": "public",
    "sdkVersion": "31.0.0",
    "ios": {
      "bundleIdentifier": "com.reactNative.sample",
      "supportsTablet": true
    },
    "android": {
      "package": "com.reactNative.sample"
    },
    "platforms": [
      "ios",
      "android"
    ],
    "version": "0.4.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "description": ""
  }
}