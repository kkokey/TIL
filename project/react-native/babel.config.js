module.exports = function(api) {
  api.cache(true);
  return {
    presets: [
      ['module:metro-react-native-babel-preset'],
      'babel-preset-expo'
    ],
    plugins: [
      '@babel/plugin-transform-runtime',
      ['@babel/plugin-transform-flow-strip-types'],
      ['@babel/plugin-proposal-decorators', { legacy: true }],
      ['@babel/plugin-proposal-class-properties', { loose: true }]
    ],
  };
};
