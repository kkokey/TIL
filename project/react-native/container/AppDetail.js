import React from 'react';
import {
  ScrollView,
  StyleSheet,
  SafeAreaView,
  View,
  Text,
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f2f2f2',
  },
  main: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  btnAlign: {
    flexDirection: 'row',
  }
});

class AppMain extends React.Component {
  static navigationOptions = () => ({
    title: 'Detail',
  });

  constructor(props) {
    super(props);
    this.state = {
      listStore: [],
      isModalVisible: false
    };
  }

  render() {
    // const navigation = this.props.navigation;
    return (
      <SafeAreaView style={styles.container}>
        <ScrollView style={styles.container}>
          <View>
            <Text>`{this.props.navigation.state.params.selectItemName}`을 선택하셨습니다! :)</Text>
          </View>
        </ScrollView>
      </SafeAreaView>
    );
  }
}

export default AppMain;
