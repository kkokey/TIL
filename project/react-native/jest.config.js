// jest.config.js
module.exports = {
  preset: "react-native",
  transform: {
    "^.+\\.jsx?$": "<rootDir>/node_modules/babel-jest"
  },
  transformIgnorePatterns: [
    "/node_modules/(?!react-native)/.+"
  ],
  testRegex: "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
  moduleFileExtensions: ["ts", "tsx", "js", "jsx", "json", "node"]
};
