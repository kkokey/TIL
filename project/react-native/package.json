{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "eject": "expo eject",
    "test": "node ./node_modules/jest/bin/jest.js --watchAll"
  },
  "jest": {
    "transform": {}
  },
  "dependencies": {
    "axios": "^0.18.0",
    "babel-cli": "^6.26.0",
    "babel-jest": "^23.6.0",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react-native": "^4.0.1",
    "eslint": "^5.10.0",
    "eslint-plugin-react": "^7.11.1",
    "expo": "^46.0.14",
    "jest": "^23.6.0",
    "jest-expo": "^31.0.0",
    "metro-react-native-babel-preset": "^0.48.5",
    "mobx": "^5.8.0",
    "mobx-react": "^5.4.3",
    "qrcode-reader": "^1.0.4",
    "react": "16.5.0",
    "react-dom": "^16.6.3",
    "react-native": "^0.57.8",
    "react-native-action-button": "^2.8.5",
    "react-native-modal": "^7.0.2",
    "react-native-platform-touchable": "^1.1.1",
    "react-navigation": "^3.0.8",
    "react-test-renderer": "^16.6.3"
  },
  "devDependencies": {
    "@babel/core": "^7.2.2",
    "@babel/preset-flow": "^7.0.0",
    "babel-core": "^7.0.0-bridge.0",
    "babel-preset-expo": "^5.0.0",
    "regenerator-runtime": "^0.13.1"
  },
  "private": true
}
