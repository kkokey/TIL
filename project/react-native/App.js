import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { createAppContainer } from "react-navigation";
import { Provider } from 'mobx-react';
import AppNavigator from './navigation/Navigator';
import stores from "./stores";

const AppContainer = createAppContainer(AppNavigator);

export default class App extends React.Component {
  render() {
    return (
      <Provider {...stores}>
        <AppContainer
          ref={nav => {
            this.navigator = nav;
          }}
        />
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
