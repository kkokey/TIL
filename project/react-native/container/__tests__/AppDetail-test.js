import 'react-native';
import React from 'react';
import AppDetail from '../AppDetail';
import renderer from 'react-test-renderer';

describe('App Detail screen', () => {
  it('should renders AppDetail', () => {
    const detailRenderJSON = renderer.create(<AppDetail/>).toJSON();
    expect(detailRenderJSON).toMatchSnapshot();
  });
});
