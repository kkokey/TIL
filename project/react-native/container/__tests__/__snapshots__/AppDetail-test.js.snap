// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Detail screen should renders AppDetail 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#f2f2f2",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#f2f2f2",
        "flex": 1,
      }
    }
  >
    <View>
      <View>
        <Text>
          Detail
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
